pymc/model.py

```diff
@@ -2035,10 +2035,74 @@ def Deterministic(name, var, model=None, dims=None):
+      When the model is sampled, the posterior distribution of ``max_items`` given the observed value of ``n_items`` will be influenced by the power-law prior defined in the Potential object
+
+     .. code:: python
+
+        with pm.Model():
+          # p(max_items) = 1 / max_items
+          max_items = pm.Uniform("max_items", lower=1, upper=100)
+          pm.Potential('power_prior', pm.math.log(1/max_items))
+
+          n_items = pm.Uniform('n_items', lower=1, upper=max_items, observed=60)
+
+      In the next example, the soft_sum_constraint potential encourages x and y to have a small sum, effectively adding a soft constraint on the relationship between the two variables.
+      This can be useful in cases where you want to ensure that the sum of multiple variables stays within a certain range, without enforcing an exact value.
+      In this case, the larger the deviation, larger will be the negative value (-((x + y)**2)) which the MCMC sampler will attempt to minimize. However, the sampler might generate values for some small deviations but with lower probability hence this is a soft constraint.
+
+     .. code:: python
+
+        with pm.Model() as model:
+          x = pm.Normal('x', mu=0.1, sigma=1)
+          y = pm.Normal('y', mu=x, sigma=1, observed=data)
+          soft_sum_constraint = pm.Potential("soft_sum_constraint", -((x + y)**2))
+
+      The potential value is incorporated into the model log-probability, so it should be -inf (or very negative) when a constraint is violated, so that those draws are rejected. 0 won't have any effect and positive values will make the proposals more likely to be accepted
+
+     .. warning::
+
+        Potential functions only influence logp based sampling, like the one used by ``pm.sample``. Potentials, modify the log-probability of the model by adding a contribution to the logp which is used by sampling algorithms which rely on the information about the observed data to generate posterior samples.
+        Potentials are not applicable in the context of forward sampling because they don't affect the prior distribution itself, only the computation of the logp.
+        Forward sampling algorithms generate sample points from the prior distribution of the model, without taking into account the likelihood function. In other words, it does not use the information about the observed data.
+        Hence, Potentials do not affect forward sampling, which is used by ``sample_prior_predictive`` and ``sample_posterior_predictive``. A warning saying "The effect of Potentials on other parameters is ignored during prior predictive sampling" is always raised to alert user of this.
+
+
     Parameters
     ----------
     name: str
-    var: PyTensor variables
+    var: PyTensor expression to be added to the model joint logp
+    model: The model object to which the potential function is added. If ``None`` is provided, the current model is used.
```

Comment:
```suggestion
    var : PyTensor expression
        Expression to be added to the model joint logp
    model : Model, optional
        The model object to which the potential function is added. If ``None`` is provided, the current model is used.
```

a space is needed both before and after the colon (existing docstrings are generally not a good example to follow), and the first line should contain only `name : type` the description comes in the 2nd+ line and indented.